src/model/error.rs

```diff
@@ -51,6 +52,66 @@ pub enum RdpErrorKind {
     UnexpectedType,
 }
 
+/// ProtocolNegFailureCode defines the failure codes
+/// for the RDP Negotiation Failure message (2.2.1.2.2).
+/// It is sent by the server to inform the client of a failure
+/// that has occurred while preparing security for the connection.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ProtocolNegFailureCode {
+    Unknown(u32),
+    SslRequiredByServer,
+    SslNotAllowedByServer,
+    SslCertNotOnServer,
+    InconsistentFlags,
+    HybridRequiredByServer,
+    SslWithUserAuthRequiredByServer,
+}
+
+impl ProtocolNegFailureCode {
+    pub fn from_code(code: u32) -> Self {
+        match code {
+            1 => ProtocolNegFailureCode::SslRequiredByServer,
```

Comment:
I think these need to be `0x1`, `0x2`, etc. to match https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/1b3920e7-0116-4345-bc45-f2c4ad012761

Reply:
Nope, hexadecimal 1 is still 1.

`0x1` is the same as `1` is the same as `0o1` is the same as `0b1`.

Reply:
Oh yeah, duh.